tests/test_result_conversion.rs

```diff
@@ -0,0 +1,45 @@
+//! Testing https://github.com/PyO3/pyo3/issues/1106. A result type that
+//! implements `From<MyError> for PyErr` should be automatically converted
+//! when using `#[pyfunction]`.
+//!
+//! This only makes sure that valid `Result` types do work. For an invalid
+//! enum type, see `ui/invalid_result_conversion.py`.
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+
+use std::fmt;
+
+mod common;
+
+/// A basic error type for the tests.
+#[derive(Debug)]
+struct MyError {
+    pub descr: String,
+}
+
+impl fmt::Display for MyError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "My error message: {}", self.descr)
+    }
+}
+
+/// Important for the automatic conversion to `PyErr`.
+impl From<MyError> for PyErr {
+    fn from(err: MyError) -> pyo3::PyErr {
+        pyo3::exceptions::PyOSError::py_err(err.to_string())
```

Comment:
It looks like that `MyError` does not have `to_string`

Reply:
Doesn't `to_string` come from `impl fmt::Display for MyError`? The test compiles so I assume `to_string` does exist.

Reply:
Pretty much, it comes from `std::string::ToString` which has a blanket impl for `T: Display`. Also, `ToString` is in the prelude, so this function is available nearly everywhere.

https://doc.rust-lang.org/std/string/trait.ToString.html#implementors

Reply:
Ah, thanks, I missed that.